code/modules/client/preferences.dm

```diff
@@ -1405,6 +1404,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	return 1
 
 /datum/preferences/proc/copy_to(mob/living/carbon/human/character, icon_updates = 1)
+	if(!generated_char_traits)
+		generated_char_traits = TRUE //usually it would be better to have a check for character_traits.len but that doesn't FUCKING WORK and I don't know why.
+		character_traits = all_traits

```

Comment:
Isn't len being phased out for length() anyway?

Reply:
If so, I've not heard of it until now.

Reply:
It's a BYOND thing, but it's been a 'thing' for a few years now. I think it says it in the reference guide.
But, does it work with length()?

Reply:
I checked the reference guide and didn't see any particular pointers about `.len` being inferior to `length()`, and I've used `.len` for a really long time, to the point where I've never actually used `length()` to check a list. I imagine it would work, though.